src/SourceGenerators/Uno.UI.SourceGenerators/Content/Uno.UI.SourceGenerators.props

```diff
@@ -371,27 +371,51 @@
 	</Target>
 
 	<Target Name="_InjectAdditionalFiles" BeforeTargets="GenerateMSBuildEditorConfigFileShouldRun;GenerateMSBuildEditorConfigFileCore" DependsOnTargets="_FillSourceGeneratorItemGroups">
-		<ItemGroup>
-			<AdditionalFiles Include="@(Page)" SourceItemGroup="Page" />
-			<AdditionalFiles Include="@(ApplicationDefinition)" SourceItemGroup="ApplicationDefinition" />
-			<AdditionalFiles Include="@(PRIResource)" SourceItemGroup="PRIResource" />
-			<AdditionalFiles Include="@(TSBindingAssemblySource)" SourceItemGroup="TSBindingAssemblySource" />
-			<AdditionalFiles Include="@(UnoPage)" SourceItemGroup="Page" />
-			<AdditionalFiles Include="@(UnoApplicationDefinition)" SourceItemGroup="ApplicationDefinition" />
-
-			<_AdditionalFilesCleanup Include="@(AdditionalFiles)" />
-			<AdditionalFiles Remove="@(_AdditionalFilesCleanup)" />
-			<AdditionalFiles Condition="'@(_AdditionalFilesCleanup)'!=''" Include="@(_AdditionalFilesCleanup->'$([System.IO.Path]::GetFullPath('%(_AdditionalFilesCleanup.Identity)'))')" />
-		</ItemGroup>
+
+    <PropertyGroup>
+      <_Pages>@(Page)</_Pages>
+      <_ApplicationDefinitions>@(ApplicationDefinition)</_ApplicationDefinitions>
+      <_PRIResources>@(PRIResource)</_PRIResources>
+      <_TSBindingAssemblySources>@(TSBindingAssemblySource)</_TSBindingAssemblySources>
+      <_UnoPages>@(UnoPage)</_UnoPages>
+      <_UnoApplicationDefinitions>@(UnoApplicationDefinition)</_UnoApplicationDefinitions>
+      <_AdditionalFiles>@(AdditionalFiles)</_AdditionalFiles>
+    </PropertyGroup>
+
+    <ItemGroup Label="Add SourceItemGroup metadata to existing additional files">
+      <AdditionalFiles SourceItemGroup="Page" Condition="$(_UnoPages) == '' AND $(_Pages) != '' AND $(_AdditionalFiles) != '' AND $(_Pages.Contains(%(AdditionalFiles.FullPath)))" />
```

Comment:
Would you be able to add a regression test in the runtimetests csproj (for instance), which create a custom `AdditionalFile` item, then validates that after the build, it's still set properly (with the right metadata)? Adding two targets, one beforebuild and one after `compile` that fails the build, would make it easier to ensure we're not breaking later on. Thanks!

Reply:
I want to note that the syntax for updating metadata through Condition looks erroneous if you don’t know about it (because you expect an explicit Update, but I couldn’t make code through Update at the time of the PR, although I think it’s possible, I just need something like this filter the collection). Here are some examples that this is a valid code and it works:
https://github.com/MicrosoftDocs/visualstudio-docs/blob/main/docs/msbuild/msbuild-items.md#updating-metadata-on-items-in-an-itemgroup-of-a-target
https://github.com/dotnet/msbuild/issues/1618#issuecomment-275691043